compiler/docgen.nim

```diff
@@ -835,15 +835,15 @@ proc docstringSummary(rstText: string): string =
   result = rstText.substr(2).strip
   var pos = result.find('\L')
   if pos > 0:
-    result.delete(pos, result.len - 1)
+    result = result[0 .. pos]
```

Comment:
buggy patch and allocates un-necessarily

Reply:
My mistake on the allocation. Slices can be done in place, so my mistake on that. Can you elaborate on "buggy"? It should achieve the same result as far as I can tell.

Reply:
off by 1 error, try it

Reply:
Thank you. I came from python, and read up on the differences. Thank you for your patience.

Reply:
why not simply use `result.setLen(pos - 1)` ?

Reply:
That does in fact do that both in place, and do the same thing. Thank you again for helping me learn about these things, and being patient.